oak/server/buffer_channel.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OAK_SERVER_BUFFER_CHANNEL_H_
+#define OAK_SERVER_BUFFER_CHANNEL_H_
+
+#include "asylo/util/logging.h"
+#include "oak/server/channel.h"
+
+namespace oak {
+
+// A channel implementation that forwards read and writes to underlying data structures, keeping
+// track of read and write cursors.
+class BufferChannel final : public Channel {
```

Comment:
BufferChannel/PrintChannel seem like verbs... maybe BufferedChannel (Stream)?

Reply:
Renamed to BufferedChannel